tests/impl.cpp

```diff
@@ -2130,6 +2134,115 @@ bool test_mm_test_all_zeros(const int32_t *_a, const int32_t *_mask)
     return result == ret ? true : false;
 }
 
+static inline uint64_t MUL(uint32_t a, uint32_t b)
+{
+    return (uint64_t)a * (uint64_t)b;
+}
+
+// From BearSSL. Performs a 32-bit->64-bit carryless/polynomial
+// long multiply.
+//
+// This implementation was chosen because it is reasonably fast
+// without a lookup table or branching.
+//
+// This does it by splitting up the bits in a way that they
+// would not carry, then combine them together with xor (a
+// carryless add).
+//
+// https://www.bearssl.org/gitweb/?p=BearSSL;a=blob;f=src/hash/ghash_ctmul.c;h=3623202;hb=5f045c7#l164
+static uint64_t clmul_32(uint32_t x, uint32_t y)
+{
+    uint32_t x0, x1, x2, x3;
+    uint32_t y0, y1, y2, y3;
+    uint64_t z0, z1, z2, z3;
+
+    x0 = x & (uint32_t)0x11111111;
+    x1 = x & (uint32_t)0x22222222;
+    x2 = x & (uint32_t)0x44444444;
+    x3 = x & (uint32_t)0x88888888;
+    y0 = y & (uint32_t)0x11111111;
+    y1 = y & (uint32_t)0x22222222;
+    y2 = y & (uint32_t)0x44444444;
+    y3 = y & (uint32_t)0x88888888;
+    z0 = MUL(x0, y0) ^ MUL(x1, y3) ^ MUL(x2, y2) ^ MUL(x3, y1);
+    z1 = MUL(x0, y1) ^ MUL(x1, y0) ^ MUL(x2, y3) ^ MUL(x3, y2);
+    z2 = MUL(x0, y2) ^ MUL(x1, y1) ^ MUL(x2, y0) ^ MUL(x3, y3);
+    z3 = MUL(x0, y3) ^ MUL(x1, y2) ^ MUL(x2, y1) ^ MUL(x3, y0);
+    z0 &= (uint64_t)0x1111111111111111;
+    z1 &= (uint64_t)0x2222222222222222;
+    z2 &= (uint64_t)0x4444444444444444;
+    z3 &= (uint64_t)0x8888888888888888;
+    return z0 | z1 | z2 | z3;
+}
+
+// Performs a 64x64->128-bit carryless/polynomial long
+// multiply, using the above routine to calculate the
+// subproducts needed for the full-size multiply.
+//
+// This uses the Karatsuba algorithm.
+//
+// Normally, the Karatsuba algorithm isn't beneficial
+// until very large numbers due to carry tracking and
+// multiplication being relatively cheap.
```

Comment:
`until very large number` is weird.
Change `cheap` to `low cost` and refactor the comment.